feat(application-system): delegations (#6645)
island-is/island.is: handbook/reference/problems/actor-validation-failed.md

# 400 Actor Validation Failed

The server will not process the request due to validation issues around actor delegations.

The client can try again after delegating to correct delegation.

## Extra metadata

### `fields`

> A JSON object listing validation issues where delegatedUser is the nationalId of the user assigned to the resource that the actor is a and actor is the nationalId of the user trying to access the resource.

Reviewer: > resource that the **actor is a and** actor is the nationalId of the user trying to access the resource.
Don't understand this. Is something missing?

Reviewer 2: @alexandradilja perhaps? :D

# 400 Actor Validation Failed

The server will not process the request due to validation issues around actor delegations.

> This error should be thrown when user (actor) tries to access a resource that they do not have access to but someone they have delegation rights for (subject) has access to.

Reviewer: ```suggestion
This error should be thrown when a user (actor) tries to access a resource that they do not have access to but someone they have delegation rights for (subject) has access to.
```

The error should contain the national id of the subject that is assigned to the resource and the types of delegations the actor has for the subject.

> Handeling this error could then involve allowing the user to delegate to the subject assigned to the resource so that they can access it as an actor.

Reviewer: ```suggestion
Handling this error could then involve allowing the user to delegate to the subject assigned to the resource so that they can access it as an actor.
```

### `fields`

A JSON object where delegatedUser is the national id of the subject that is assigned to the resource that the actor is trying to access.
The delegations array contains delegations that the actor has for the subject assigned to the resource.

> The delegation object contain the name of the subject, the subjects national id as well as the type of delegation the actor has for the subject.

Reviewer: ```suggestion
The delegation object contains the name of the subject, the subject's national id as well as the type of delegation the actor has for the subject.
```
